src/main/scala/com/github/unchama/seichiassist/menus/home/HomeMenu.scala

```diff
@@ -131,61 +136,19 @@ case class HomeMenu(pageIndex: Int = 0) extends Menu {
           )
         }
       )
-
-    def setHomeButton(homeNumber: Int)(implicit environment: Environment): Button =
-      Button(
-        new IconItemStackBuilder(Material.BED)
-          .title(s"$YELLOW$UNDERLINE${BOLD}ホームポイント${homeNumber}を設定")
-          .lore(
-            List(
-              s"${GRAY}現在位置をホームポイント$homeNumber",
-              s"${GRAY}として設定します",
-              s"$DARK_GRAY※確認メニューが開きます",
-              s"$DARK_RED${UNDERLINE}クリックで設定",
-              s"${DARK_GRAY}command->[/home set $homeNumber]"
-            )
-          )
-          .build(),
-        LeftClickButtonEffect {
-          SequentialEffect(
-            FocusedSoundEffect(Sound.BLOCK_FENCE_GATE_OPEN, 1f, 0.1f),
-            environment.ioCanOpenConfirmationMenu.open(HomeChangeConfirmationMenu(homeNumber))
-          )
-        }
-      )
-
-    def removeHomeButton(homeNumber: Int)(implicit environment: Environment): Button =
-      Button(
-        new IconItemStackBuilder(Material.WOOL, 14)
-          .title(s"$RED$UNDERLINE${BOLD}ホームポイント${homeNumber}を削除")
-          .lore(
-            List(
-              s"${GRAY}ホームポイント${homeNumber}を削除します。",
-              s"$DARK_GRAY※確認メニューが開きます。",
-              s"$DARK_RED${UNDERLINE}クリックで設定"
-            )
-          )
-          .build(),
-        LeftClickButtonEffect {
-          FocusedSoundEffect(Sound.BLOCK_ENDERCHEST_CLOSE, 1f, 0.1f)
-          SequentialEffect(
-            environment
-              .ioCanOpenHomeRemoveConfirmationMenu
-              .open(HomeRemoveConfirmationMenu(homeNumber))
-          )
-        }
-      )
   }
 }
-case class HomeMenuButtonComputations(player: Player) {
-  def setHomeNameButton[F[_]: HomeReadAPI: ConcurrentEffect](homeNumber: Int): IO[Button] = {
-    import cats.effect.implicits._
-    import cats.implicits._
+case class HomeMenuButtonComputations(player: Player)(
+  private implicit val environment: HomeMenu.Environment
+) {
+  import cats.effect.implicits._
```

Comment:
```suggestion
  import cats.effect.implicits._
  import environment._
```
とすると、`homeReadAPI`を呼び出す際に`environment`をチェーンしなくて良くなります